### ISSUE TITLE
Quantum armor flight bug on servers only seems like

### ISSUE THREAD

**Issue by PurelyRain** [TechReborn/TechReborn#3186]
Playing tech reborn for 1.20.1 on a fabric server, I have tested quantum armor flight in both LAN and single player worlds but no matter if its a custom server hosted on my pc with a server file or a paid for one (im playing on a paid server with my friends) the flight takes a solid minute to activate, it does one of these:
lets you hover slightly by i assume turning your flight on and off because the screen becomes buggy as if the FOV keeps being changed
lets you hover a bit "better" but the FOV changing effect becomes "faster"
actually lets you fly 
now, it is random on how many times i have to press the space bar to actually fly, sometimes its first try and sometimes i have to struggle for 2 minutes to have flight properly activate, i have looked through configs and such to see if anything could be causing this but i have not gotten anywhere.
thanks for any help in advance!

**Comment by drcrazy** [comment]
Do you have other mods installed besides TR?

**Comment by PurelyRain** [comment]
> Do you have other mods installed besides TR?

Yes I do, I believe there were nearly 100 mods? But the thing is that the mods don't change, there's no mods that are only server side or only client side, so I don't see why they would matter when the bug persists only on servers, I even tried with just TR and it still kept happening.

**Comment by drcrazy** [comment]
So it's a LAN with just a TR, right?

**Comment by PurelyRain** [comment]
> So it's a LAN with just a TR, right?

Nah it's a paid server where the bug happens, LAN and single player is fine tho

**Comment by drcrazy** [comment]
Then in could be something on that server which mess with either this:
https://github.com/TechReborn/TechReborn/blob/3ef8b87412086d009982e5715e0e3f2392771e89/src/main/java/techreborn/items/armor/QuantumSuitItem.java#L94

or that: 
https://github.com/TechReborn/TechReborn/blob/3ef8b87412086d009982e5715e0e3f2392771e89/src/main/java/techreborn/items/armor/QuantumSuitItem.java#L99

**Comment by drcrazy** [comment]
And both of these are vanilla methods :(

**Comment by Trent2087** [comment]
This is an issue with playerAbillityLib. In 1.19 and previously, PAL would just spam the console and not break flight. Now it 1.20.1 it breaks TR Quantum Armor Flight, while also spamming the console infinitely.

**Comment by burdoto** [comment]
This may be related; I noticed at some point our Fabric server started spitting out [this](https://pastebin.com/GGBDHBRE) exception over and over again. Server did not crash and functionality was not notably affected, yet it keeps going permanently.

**Comment by burdoto** [comment]
Can this be resolved at some point? It makes ALL kinds of logs unreadable and bloats log size infinitely.

**Comment by burdoto** [comment]
Today, I (yet again) was facing a 1GB log file that consists mostly of this trash message.
You should know just as well as me that especially printing messages takes up quite a big chunk of performance, and this bug has a noticeable effect on performance on a larger scale.

Over 8 million lines of log file, just for some useless, nonsense warning.

Please resolve this. Otherwise our last resort will be to banish TR from our modpacks.

**Comment by Trent2087** [comment]
Someone fixed this for them. This mod adds Player Abillity Lib support to TR
https://www.curseforge.com/minecraft/mc-mods/techreborn-player-ability-lib-compat/files/5275410

**Comment by burdoto** [comment]
Thank you for pointing that out, @Trent2087. I will look into this.

**Comment by ProtectYourIP** [comment]
This worked for me too

**Comment by DreiPixel** [comment]
any fix for this yet?
im on 1.21.1 and i dont even have playerabiltyLib but i get the same console spam.  its not even uploaded for 1.21.1 yet
i think another mod i uses provides it but im not sure.

**Comment by ZurrTum** [comment]
@DreiPixel 
As long as there is a mod that contains playerabilitylib, it will cause players to be unable to fly.
### The following pull requests should be compatible:
- #3401 

But this approach is not the best way.
I don't know when the fabric API will support similar features.

**Comment by DreiPixel** [comment]
@ZurrTum 
i tested that fix. it works on servers but it will break singleplayer because theres no check for !world.isClient and PlayerAbilityLib will crash the game if it is used on a Singleplayer instance.
(i did made some changes to that for me to include a simple !world.isClient

https://github.com/Ladysnake/PlayerAbilityLib/blob/4da7e89828ffae97203d68c1dd8766111d431cc5/src/main/java/io/github/ladysnake/pal/impl/PlayerAbilityView.java#L46
